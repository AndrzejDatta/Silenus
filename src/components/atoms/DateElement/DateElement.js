import React from "react";
import PropTypes from "prop-types";
import { StyledDate, StyledDateText } from "./DateElement.styles.";
const DateElement = ({ dayName = "Mon", dayNumber = 21, isNow = false }) => {
  return (
    <StyledDate isNow={isNow}>
      <StyledDateText isNow={isNow}>{dayName}</StyledDateText>
      <StyledDateText isNumber isNow={isNow}>
        {dayNumber}
      </StyledDateText>
    </StyledDate>
  );
};

DateElement.propTypes = {
  /**name of day */
  dayName: PropTypes.string.isRequired,
  /**number of day */
  dayNumber: PropTypes.number.isRequired,
  /**bool which decides about background color */
  isNow: PropTypes.bool.isRequired,
};

export default DateElement;
