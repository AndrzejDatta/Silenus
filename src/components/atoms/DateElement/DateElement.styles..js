import styled from "styled-components";

export const StyledDate = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  background-color: ${(props) =>
    props.isClicked || props.isToday
      ? ({ theme }) => theme.color.fontGreen
      : ({ theme }) => theme.color.lightGrey};
  border-radius: 8px;
  width: 40px;
  height: 56px;
`;
export const StyledDateText = styled.p`
  letter-spacing: 1px;
  font-size: ${({ theme }) => theme.font.size.medium};
  color: ${(props) =>
    props.isClicked || props.isToday
      ? "white"
      : ({ theme }) => theme.color.fontGreen};
  font-family: ${({ theme }) => theme.font.family.poppins};
  font-weight: ${(props) => (props.isNumber ? "bold" : "normal")};
`;
