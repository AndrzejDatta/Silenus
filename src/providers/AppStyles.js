import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
