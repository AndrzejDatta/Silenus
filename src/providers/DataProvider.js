import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
