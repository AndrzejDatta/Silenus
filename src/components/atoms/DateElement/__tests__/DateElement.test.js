import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen, rerender } from "@testing-library/react";
import DateElement from "../DateElement";

describe("Test DateElement Component", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <DateElement />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("background-color", "#F9F9F9");
  });
  it("Test numbers style", () => {
    render(
      <AppStyles>
        <DateElement isNumber={true} dayNumber={14} />
      </AppStyles>
    );
    expect(screen.getByText(/14/i)).toHaveStyleRule("font-weight", "bold");
  });
});
