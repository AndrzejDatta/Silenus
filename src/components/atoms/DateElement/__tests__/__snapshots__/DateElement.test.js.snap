// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateElement Component Test default style 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F9F9F9;
  border-radius: 8px;
  width: 40px;
  height: 56px;
}

.c1 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 14px;
  color: #3AA688;
  font-family: "Poppins",sans-serif;
  font-weight: normal;
}

.c2 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 14px;
  color: #3AA688;
  font-family: "Poppins",sans-serif;
  font-weight: bold;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    Mon
  </p>
  <p
    className="c2"
  >
    21
  </p>
</div>
`;
